### ISSUE TITLE
I am trying to create a model with multiple outputs (4) with different shapes. I created a custom loss function with (y_true, y_pred) parameters and I expected that I will recieve a list of all outputs as y_pred. But instead I get only one of the output as y_pred.   

### ISSUE THREAD

**Issue by manastahir** [keras-team/keras#14140]
**Important note:** the development of Keras is currently taking place at [`github.com/tensorflow/tensorflow`](https://github.com/tensorflow/tensorflow). Please file your bug report by creating a new issue in the TensorFlow repository.

Please note that multi-backend Keras development has been discontinued. Do not report issues about multi-backend Keras (Keras 2.3.1 and lower), only report issues about the TensorFlow implementation of Keras (tf.keras).

**Comment by manastahir** [comment]
How can I receive a list of all outputs as input to my loss fucntion ?

**Comment by Vincent9797** [comment]
Could you give an example of your code?

**Comment by s-hood** [comment]
I have this question too. Some example code:
```
model = Model(
        inputs=[input_a, input_b],
        outputs=[output_a, output_b, output_c, output_d],
    )

def loss(y_true, y_pred):
    # i want to get the 4 outputs of the model as y_pred, as they are all needed to calc loss
    assert len(y_pred) == 4  # fails
```

**Comment by Vincent9797** [comment]
Could you show how you are compiling the model? If you instantiate the Model object with a list of outputs, you will need a list of loss functions when compiling it. This is fine if you are calculating each of their loss individually and you do not to do any loss calculation involving ouput_a and output_b.

An example is below:
```
model = Model(
    inputs=[input_a, input_b],
    outputs=[output_a, output_b, output_c, output_d],
)

model.compile(optimizer=..., loss = ['rmse']*4)
```

However, if your loss calculation requires multiple outputs to be used, what can do is add a concatenate layer at the end of your model to produce one single output. From there, you should be able to access the 4 outputs in your custom loss function.


An example is below:
```
output_a = # some layer
output_b = # some layer
output_c = # some layer
output_d = # some layer
output = tf.keras.backend.concatenate([output_a, output_b, output_c, output_d])
model = Model(
    inputs=[input_a, input_b],
    outputs=output,
)

def custom_loss(y_true, y_pred):
    # i want to get the 4 outputs of the model as y_pred, as they are all needed to calc loss
    output_a, output_b, output_c, output_d = y_pred
    # calculate loss

model.compile(optimizer=..., loss = custom_loss) # use the custom loss that you have created
```

**Comment by QColeman97** [comment]
@Vincent9797 You stunning, beautiful human being. Sometimes the problems with the simplest solutions are the ones you spend literally months with wrangling 😭😭😭

**Comment by leokster** [comment]
I have the same problem, and in my case, it is not possible to concatenate at the end since the different output tensors have different shapes. So, I would really need a loss function that gets the list of outputs as y_pred input.

**Comment by manastahir** [comment]
For this you have to write loss functions seperately. If you are using tensorflow.keras you can use gradient tape to make your life easy.

**Comment by manaspalaparthi** [comment]
I have the same problem too.  my model outputs 3 tensors. 
 `model= Model(inputs = inputs,outputs =[predicted_image,depth,x_train])`   

my loss function should be  

`
def depth_loss_function(y_true, y_pred):
    print(y_pred)  # ERROR  only one tensor.

    predict_image, depth , input_image = y_pred   # ERROR   To iterate over this tensor use tf.map_fn.

    # Point-wise depth
    depth_loss = K.mean(K.abs(depth - y_true), axis=-1)

    # image reconstruction
    mse = tf.keras.losses.MeanSquaredError()
    image_loss = mse(input_image - predict_image)

    # Weights
    w1 = 1.0
    w2 = 1.0

    return  (w2 * depth_loss) + (w2 * image_loss )`

compile method 

`model.compile(optimizer= optimizer, loss =depth_loss_function)`

**Comment by leokster** [comment]
@manaspalaparthi maybe you can also stress this out in the issue I opened a few days ago: https://github.com/tensorflow/tensorflow/issues/46428

**Comment by vladhondru25** [comment]
Has anyone found a solution to this? I am encountering the same issue: I have a model with two outputs (different shapes so I cannot concatenate them) and I need to pass them together in the loss function.

**Comment by abdulbasitds** [comment]
I have the same issue, I have 3 outputs but custom loss function gets only one of them

**Comment by gadagashwini** [comment]
@manastahir, 
Take a look at workaround mention on similar [thread](https://stackoverflow.com/a/67803627/14290244). Let us know if it helps. Thanks!
